dev-embedded/sunxi-tools/sunxi-tools-1.4.2.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+
+inherit toolchain-funcs
+
+if [[ ${PV} = *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/linux-sunxi/sunxi-tools"
+else
+	KEYWORDS="~amd64"
+	SRC_URI="https://github.com/linux-sunxi/sunxi-tools/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+fi
+
+DESCRIPTION="A collection of command line tools for ARM devices with Allwinner SoCs"
+HOMEPAGE="https://linux-sunxi.org/Main_Page"
+
+LICENSE="GPL-2"
+SLOT="0"
+
+RDEPEND="acct-group/plugdev
+	virtual/libusb:1
+	virtual/udev"
+
+DEPEND="${RDEPEND}
+"
+
+BDEPEND="virtual/pkgconfig"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-1.4.1-fix-strncpy-compiler-warning.patch"
+)
+
+src_prepare() {
+	# Substitute test URL
+	# 1.4.2 was released in 2016, therefore it cannot be tested against master
+	sed -i 's$^BOARDS_URL.*$BOARDS_URL := https://github.com/linux-sunxi/sunxi-boards/archive/bc7410fed9e5d9b31cd1d6ae90462d06b513660e.zip$' tests/Makefile || die
+	sed -i 's$sunxi-boards-master$sunxi-boards-bc7410fed9e5d9b31cd1d6ae90462d06b513660e$' tests/Makefile || die
```

Comment:
Thank you for fixing the tests, but now we have another problem: tests that access the network [are forbidden](https://devmanual.gentoo.org/ebuild-writing/functions/src_test/index.html#tests-that-require-network-or-service-access).  They fail if run with the sandbox active:
```
>>> Test phase: dev-embedded/sunxi-tools-1.4.2
make -j12   check
make -C tests/
make[1]: Entering directory '/mnt/tmp/portage/dev-embedded/sunxi-tools-1.4.2/work/sunxi-tools-1.4.2/tests'
curl -fLsS -o sunxi-boards.zip https://github.com/linux-sunxi/sunxi-boards/archive/bc7410fed9e5d9b31cd1d6ae90462d06b513660e.zip
cc -Wall -Werror -o unify-fex unify-fex.c
# Usage help / invocation with no args
../sunxi-fexc -? 2> /dev/null ; exit 0
# Improve code coverage for corner cases (e.g. erroneous parameters)
./test_fex2bin_corner_cases.sh
./test_fex2bin_corner_cases.sh: line 8: warning: command substitution: ignored null byte in input
./test_bin2fex_corner_cases.sh
E: fexc: sunxi-boards/sys_config/a10/a10-olinuxino-lime.fex: No such file or directory
cat: sunxi-boards/sys_config/a10/a10-olinuxino-lime.bin: No such file or directory
fexc-bin: /dev/stdin: version: 0.0
fexc-bin: /dev/stdin: size: 0 (0 sections), header value: 0
curl: (6) Could not resolve host: github.com
make[1]: *** [Makefile:23: sunxi-boards.zip] Error 6
make[1]: Leaving directory '/mnt/tmp/portage/dev-embedded/sunxi-tools-1.4.2/work/sunxi-tools-1.4.2/tests'
make: *** [Makefile:196: check] Error 2
 * ERROR: dev-embedded/sunxi-tools-1.4.2::gentoo failed (test phase):
 *   Make check failed. See above for details.
 * 
 * Call stack:
 *            ebuild.sh, line  136:  Called src_test
 *          environment, line 1304:  Called default
 *   phase-functions.sh, line  872:  Called default_src_test
 *   phase-functions.sh, line  901:  Called __eapi0_src_test
 *     phase-helpers.sh, line  832:  Called die
 * The specific snippet of code:
 *   		${emake_cmd} ${internal_opts} check || \
 *   			die "Make check failed. See above for details."
 * 
 * If you need support, post the output of `emerge --info '=dev-embedded/sunxi-tools-1.4.2::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=dev-embedded/sunxi-tools-1.4.2::gentoo'`.
 * The complete build log is located at '/mnt/tmp/portage/dev-embedded/sunxi-tools-1.4.2/temp/build.log'.
 * The ebuild environment file is located at '/mnt/tmp/portage/dev-embedded/sunxi-tools-1.4.2/temp/environment'.
 * Working directory: '/mnt/tmp/portage/dev-embedded/sunxi-tools-1.4.2/work/sunxi-tools-1.4.2'
 * S: '/mnt/tmp/portage/dev-embedded/sunxi-tools-1.4.2/work/sunxi-tools-1.4.2'
```

My suggestion is to add a `test` USE flag and add the zip file to `SRC_URI` conditionally, put it in `tests/` during `src_prepare` and patch the Makefile to not attempt to download it (this last step might be unnecessary, from how the makefile looks, but I didn't test).

Also please check out the [QA policy](https://projects.gentoo.org/qa/policy-guide/other-metadata.html#pg0703) about adding the `test` flag.

Reply:
This could work for 1.4.2, but what about live version? I believe it wouldn't be a good idea to fix this testing zip to a specific commit for live version.

Reply:
The live ebuild is special anyway, I think for this case we can use `RESTRICT="test"` with [PROPERTIES="test_network"](https://mgorny.pl/articles/the-ultimate-guide-to-eapi-8.html#properties-test-network-to-ease-reenabling-tests-requiring-internet)

So basically have an extra zip in `SRC_URI` for the normal version, and require networking to run tests in the live version.